llvm/test/CodeGen/AArch64/float16-promotion-with-nofp.ll

```diff
@@ -0,0 +1,21 @@
+; RUN: llc -mcpu=cortex-r82 -O1 -o - %s | FileCheck %s
+
+; Source used:
+; __fp16 f2h(float a) { return a; }
+; Compiled with: clang --target=aarch64-arm-none-eabi -march=armv8-r+nofp
+
+define hidden noundef nofpclass(nan inf) half @f2h(float noundef nofpclass(nan inf) %a) local_unnamed_addr #0 {
```

Comment:
This can probably be cleaned up a bit - Remove hidden, noundef, the fpclass and local_unnamed_addr #0